neural_lam/models/ar_model.py

```diff
@@ -597,3 +597,7 @@ def on_load_checkpoint(self, checkpoint):
         if not self.restore_opt:
             opt = self.configure_optimizers()
             checkpoint["optimizer_states"] = [opt.state_dict()]
+
+    def on_run_end(self):
```

Comment:
When is this hook called? I can't find it in the Lightning docs. I am thinking that we should save the config as soon as possible, if the training run would crash at some point. But maybe that is what this does? Otherwise I guess `on_train_start` and `on_test_start` would be good hooks for this.

Also minor thing: Add docstring

Reply:
Good idea, I switched the hooks to the ones you suggest. CHANGELOG is ready as well.